deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/Documentor.java

```diff
@@ -0,0 +1,127 @@
+package com.airbnb.deeplinkdispatch;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+
+import javax.annotation.Nullable;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.tools.Diagnostic;
+
+/**
+ * Documents deep links.
+ * <p>
+ * The output format is:
+ * {DeepLink1}\n|#|\n[Description part of java doc]\n|#|\n{ClassName}#[MethodName]\n|##|\n
+ * {DeepLink2}\n|#|\n[Description part of java doc]\n|#|\n{ClassName}#[MethodName]\n|##|\n
+ * ...
+ * <p>
+ * The output location is specified in the project's gradle file through
+ * {@link ProcessingEnvironment}'s compiler argument options by deepLinkDoc.output.
+ */
+final class Documentor {
+  private static final String PROPERTY_DELIMITER = "\\n|#|\\n";
+  private static final String ELEMENT_DELIMITER = "\\n|##|\\n";
+  private static final String CLASS_METHOD_NAME_DELIMITER = "#";
+  private static final String PARAM = "@param";
+  private static final String RETURN = "@return";
+  @VisibleForTesting static final String DOC_OUTPUT_PROPERTY_NAME = "deepLinkDoc.output";
+
+  private final ProcessingEnvironment processingEnv;
+  private final Messager messager;
+
+  @Nullable
+  private File file;
+
+  Documentor(ProcessingEnvironment processingEnv) {
+    this.processingEnv = processingEnv;
+    messager = processingEnv.getMessager();
+    initFile();
+  }
+
+  void write(List<DeepLinkAnnotatedElement> elements) {
+    if (file == null) {
+      messager.printMessage(Diagnostic.Kind.NOTE,
+          "Output file is null, DeepLink doc not generated.");
+      return;
+    }
+    if (elements == null || elements.isEmpty()) {
+      messager.printMessage(Diagnostic.Kind.NOTE,
+          "No deep link, DeepLink doc not generated.");
+      return;
+    }
+    PrintWriter writer;
+    try {
+      writer = new PrintWriter(new FileWriter(file), true);
+      for (DeepLinkAnnotatedElement element : elements) {
+        writer.print(element.getUri() + PROPERTY_DELIMITER);
+        String doc = formatJavaDoc(
+            processingEnv.getElementUtils().getDocComment(element.getElement()));
+        if (doc != null) {
+          writer.print(doc);
+        }
+        writer.print(PROPERTY_DELIMITER);
+        writer.print(element.getAnnotatedElement().getSimpleName());
+        if (element.getAnnotationType().equals(DeepLinkEntry.Type.METHOD)) {
+          writer.print(CLASS_METHOD_NAME_DELIMITER);
+          writer.print(element.getMethod());
+        }
+        writer.print(ELEMENT_DELIMITER);
+      }
+      writer.flush();
+      writer.close();
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
instead of `printStackTrace`, do something like
```
messager.printMessage(Diagnostic.Kind.ERROR,
               "DeepLink doc not generated: " + e.getMessage());
```